src/util/freer.h

```diff
@@ -0,0 +1,24 @@
+/*******************************************************************\
+
+Module:
+
+Author: Reuben Thomas, reuben.thomas@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_FREER_H
+#define CPROVER_UTIL_FREER_H
+
+#include <cstdlib>
+#include <utility>
+
+struct freert
```

Comment:
Comments, please. What is it good for, when to use it, when not to use it (if applicable). This currently has a single usage, and if that's the only place it is meaningful then there doesn't seem to be any reason to have this separate file.